src/main/java/com/adobe/abp/regola/rules/RangeRule.java

```diff
@@ -0,0 +1,304 @@
+                    r.operator = getOperator();
+                    r.description = getDescription();
+                    r.result = result;
+                    r.actualValue = evaluatedFact;
+                    r.actualValues = evaluatedFacts;
+                    r.expectedValues = expectedValues;
+                    if (!expectedValues.isEmpty()) {
+                        r.expectedValue = expectedValues.iterator().next();
+                    }
+                    r.message = message;
+                    r.cause = cause;
+                    r.ignored = isIgnore();
+                }).build();
+            }
+
+            @Override
+            public CompletableFuture<Result> status() {
+                return resolveFact(factsResolver, getKey())
+                        .thenCompose(fact -> FutureUtils.supplyAsync(() -> handleSuccess(fact), executor))
+                        .whenComplete((result, throwable) -> Optional.ofNullable(getAction())
+                                .ifPresent(action -> action.onCompletion(result, throwable, snapshot())))
+                        .exceptionally(this::handleFailure);
+            }
+
+            private Result handleSuccess(T fact) {
+                if (fact == null) {
+                    result = Result.INVALID;
+                    return result;
+                }
+
+                evaluatedFact = fact;
+                result = evaluateRange(fact);
+                return result;
+            }
+
+            private Result handleFailure(Throwable throwable) {
+                result = Result.FAILED;
+                message = throwable.getMessage();
+                cause = throwable;
+                return result;
+            }
+        };
+    }
+
+    private Result evaluateRange(T fact) {
+        switch (getOperator()) {
+            case BETWEEN:
+                return evaluateBetween(fact);
+            case IN:
+                return evaluateIn(fact);
+            case IS_BEFORE:
+                return evaluateIsBefore(fact);
+            case IS_AFTER:
+                return evaluateIsAfter(fact);
+            default:
+                return Result.OPERATION_NOT_SUPPORTED;
+        }
+    }
+
+    private Result evaluateBetween(T fact) {
+        int compareMin = fact.compareTo(min);
+        int compareMax = fact.compareTo(max);
+
+        boolean afterMin = minExclusive ? compareMin > 0 : compareMin >= 0;
+        boolean beforeMax = maxExclusive ? compareMax < 0 : compareMax <= 0;
+
+        return afterMin && beforeMax ? Result.VALID : Result.INVALID;
+    }
+
+    private Result evaluateIn(T fact) {
+        if (step == null) {
+            return evaluateBetween(fact);
+        }
+
+        if (fact instanceof Number && step instanceof Number) {
+            double value = ((Number) fact).doubleValue();
+            double minVal = ((Number) min).doubleValue();
+            double stepVal = ((Number) step).doubleValue();
```

Comment:
I am not sure I understand the `step`. Could we remove this for this initial iteration please?